Source/MLX/MLXArray+Bytes.swift

```diff
@@ -0,0 +1,314 @@
+// Copyright © 2024 Apple Inc.
+
+import Cmlx
+import Foundation
+import Metal
+
+// MARK: - Backing / Bytes
+
+extension MLXArray {
```

Comment:
This is new code + code moved from `MLXArray.swift` into this new file -- there are a number of methods dealing with bytes/backing that are consolidated here.